jmxtrans-output/jmxtrans-output-influxdb/src/main/java/com/googlecode/jmxtrans/model/output/InfluxDbWriter.java

```diff
@@ -203,6 +213,15 @@ public void doWrite(Server server, Query query, Iterable<Result> results) throws
 			resultAttribute.addAttribute(resultTagMap, result);
 		}
 
+		if(typeNamesAsTags) {
+			Map<String, String> typeNameValueMap = TypeNameValue.extractMap(resultTagMap.get("typeName"));
+			for (String typeToTag : this.typeNames) {
+				if (typeNameValueMap.containsKey(typeToTag)) {
+					resultTagMap.put(typeToTag, typeNameValueMap.get(typeToTag));
+				}
+			}
+		}
```

Comment:
Not sure if we've moved to Java8 yet, but it seems like this could be one line of code.

Reply:
I do believe we still have users on Java 7, so the move to Java 8 is [not there yet](https://github.com/jmxtrans/jmxtrans/blob/master/pom.xml#L776-L783).